### PR TITLE
Improve docstring of Axes.spy

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

As part of #10148: Docstring update for `Axes.spy`.

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -7293,61 +7293,82 @@ def specgram(self, x, NFFT=None, Fs=None, Fc=None, detrend=None,
...
+            will not be square.
 
+            This parameter is a shortcut for explicitly calling
+            `.Axes.set_aspect`. See there for further details.
 
-            If 'auto', changes the image aspect ratio to match that of the
-            axes.
+            - 'equal': Ensures an aspect ratio of 1. Pixels will be square
+              (unless pixel sizes are explicitly made non-square in data
+              coordinates using *extent*).
```
Comment (ImportanceOfBeingErnest) [inline lib/matplotlib/axes/_axes.py]: You cannot use `extent` on spy (not sure if you should, though).

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -7293,61 +7293,82 @@ def specgram(self, x, NFFT=None, Fs=None, Fc=None, detrend=None,
...
-        remaining kwargs passed to the
-        :meth:`~matplotlib.axes.Axes.plot` method.
+        Additional Parameters
+        ---------------------
+        **kwargs
+            The supported additional parameters depend on the plotting style.
 
-        If *marker* and *markersize* are *None*, useful kwargs include:
+            For the image style, you can pass any parameters supported by
+            `~.Axes.imshow`. Commonly used parameters include *cmap* and
```
Comment (ImportanceOfBeingErnest) [inline lib/matplotlib/axes/_axes.py]: Not true, you cannot pass `extent`.

File: lib/matplotlib/axes/_axes.py [hunk 3]
```diff
@@ -7293,61 +7293,82 @@ def specgram(self, x, NFFT=None, Fs=None, Fc=None, detrend=None,
...
+            *alpha*.
 
-        * *cmap*
-        * *alpha*
+            For the marker style, you can pass any parameters
+            supported by `~.Axes.plot. Commonly used parameters include
+            *color*, *marker* and *markersize*.
```
Comment (ImportanceOfBeingErnest) [inline lib/matplotlib/axes/_axes.py]: You cannot pass `linestyle`.

Also *marker* and *markersize* are already arguments of `spy`, so one wouldn't list them as "Additional Parameters" as well,right?

**Comment by timhoffm** [comment]
@meeseeksdev backport to v2.2.3

**Comment by timhoffm** [comment]
@meeseeksdev backport to v2.2.x

**Comment by adamjstewart** [comment]
It looks like the docs for [`matplotlib.pyplot.spy`](https://matplotlib.org/api/_as_gen/matplotlib.pyplot.spy.html) and [`matplotlib.axes.Axes.spy`](https://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.spy.html) are messed up. All rST formatting has been removed and most of the docs are formatted in bold. Is this why the commit needs to be backported?

**Comment by ImportanceOfBeingErnest** [comment]
Correct. But you can already view the improved version [here](https://matplotlib.org/devdocs/api/_as_gen/matplotlib.pyplot.spy.html).

**Pull request merged**